docs/MIGRATION_GUIDE_08.md

```diff
@@ -0,0 +1,238 @@
+# Migration guide matter.js Device API to 0.8
+
+With version 0.8 the matter.js project introduces a new high level API to build devices bases on code generation
+for all Matter 1.1 device types, clusters and functionality. In the past we already generated the code for all 
+cluster definitions and this got now enhanced for all device types. Additionally, the way how own cluster logic can be 
+implemented also needed to be adjusted and got enhanced together with a very flexible way to choose the wanted cluster 
+features.
+
+This means that developers need to adjust their code to use the new classes and concepts introduced by this change. The
+former API (pre 0.8, called "Legacy" for of now) is still 100% functional, but will be deprecated and removed in upcoming 
+releases and might not get cluster specific adjustments or new features if they are not part of the core library
+functionality!
+The Controller API is still the same as before, but will be also adjusted in the future to the new concepts where 
+applicable.
+
+This document tries to give an overview how the commonly used components and classes from the legacy API need to be 
+adjusted for the new API.
+
+## Examples
+Matter.js contains several examples to show how devices are built that also can be used in practice as CLI scripts. 
+These examples got also adjusted and exists for Legacy API (*Legacy.ts) and new API. This can be used too to see the 
+differences between the APIs.
+
+## Components
+The following sections shows the legacy and matching new components and tries to show the differences and what they 
+have in common.
+
+### New:Environment <--> Legacy:MatterServer
+The new API introduces an Environment singleton which represents a platform specific basic environment and is in our 
```

Comment:
Environment isn't a singleton, Environment.default is used by default but you can supply a different environment anywhere it's used

Reply:
Ok, will adjust innext PR. Already done locally